### PR TITLE
fix(reboot): ignore short uptime change before real reboot

### PR DESCRIPTION
**Pull request by amoskong**
In one job, I found the `uptime -s` result increased 1 second without real
reboot, it might be caused by normal timedrift. So we should not treat it as
reboot finish if the uptime change is very short.

This patch added a time gap check, it will ignore short uptime change before
real reboot.

Fixes: #2209 

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1075,12 +1075,17 @@ def hard_reboot(self):  # pylint: disable=no-self-use
...
         def uptime_changed():
             try:
                 result = self.remoter.run('uptime -s', ignore_status=True)
-                return pre_uptime != result.stdout
+                post_uptime = datetime.strptime(result.stdout.strip(), '%Y-%m-%d %H:%M:%S')
+                # In one job, I found the `uptime -s` result increased 1 second without real
+                # reboot, it might be caused by normal timedrift. So we should not treat it as
+                # reboot finish if the uptime change is very short.
+                # The added a time gap check will ignore short uptime change before real reboot.
+                return pre_uptime != post_uptime and (post_uptime - pre_uptime).seconds > 30
```
Comment (roydahan) [inline sdcm/cluster.py:1088]: cant we have 2 reboots in 30 seconds?

Comment (amoskong) [reply sdcm/cluster.py:1088]: The system uptime, is the time kernel starts, not the time that system or network is ready.

Currently the time gap should be gap between two reboot (between two kernel start points).
We don't have chance to reboot the system time in 30 seconds, so it's safe.

I checked one log, it costed almost 2 mins for `cloud hypervisor reset / bios reset`,  and it took additional 1 min and 27 seconds to start the system and get network ready.

So one complete reboot  (from start hard reboot to ssh is ready) requires about 3 mins and 30 seconds.


```
### check the old uptime and execute reboot
< t:2020-06-01 14:05:56,062 f:nemesis.py      l:276  c:sdcm.nemesis         p:DEBUG > sdcm.nemesis.ChaosMonkey: Rebooting 4 out of 8 times
< t:2020-06-01 14:05:56,287 f:remote.py       l:306  c:sdcm.remote          p:DEBUG > RemoteCmdRunner [centos@2a05:d018:223:f00:162a:979e:8fea:83e4]: Running command "uptime
 -s"...
< t:2020-06-01 14:05:56,353 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@2a05:d018:223:f00:162a:979e:8fea:83e4]: 2020-06-01 14:02:46

< t:2020-06-01 14:05:56,133 f:cluster.py      l:1078 c:sdcm.cluster_aws     p:DEBUG > Node longevity-10gb-3h-4-1-db-node-b371d479-4 [34.251.15.31 | 10.0.237.67 | 2a05:d018:2
23:f00:162a:979e:8fea:83e4] (seed: False): Hardly rebooting node


### Kernel started   (it costed almost 2 mins for cloud (hypervisor) / bios rest.

2020-06-01T14:05:56+00:00  longevity-10gb-3h-4-1-db-node-b371d479-4 !INFO    | systemd: Stopping NTP client/server...
2020-06-01T14:07:53+00:00  longevity-10gb-3h-4-1-db-node-b371d479-4 !NOTICE  | kernel: Linux version 5.6.15-1.el7.elrepo.x86_64 (mockbuild@Build64R7) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-39) (GCC)) #1 SMP Wed May 27 12:54:35 EDT 2020

### ssh is ready and successfully get the uptime

< t:2020-06-01 14:09:17,664 f:remote.py       l:306  c:sdcm.remote          p:DEBUG > RemoteCmdRunner [centos@2a05:d018:223:f00:162a:979e:8fea:83e4]: Running command "uptime
 -s"...
< t:2020-06-01 14:09:17,819 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@2a05:d018:223:f00:162a:979e:8fea:83e4]: 2020-06-01 14:07:50
< t:2020-06-01 14:09:17,826 f:remote.py       l:156  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@2a05:d018:223:f00:162a:979e:8fea:83e4]: Command "uptime -s" finished with status 0
```

Comment (amoskong) [reply sdcm/cluster.py:1088]: The timedrift I touched is only +1 second, I can reduce the 30 to 5 if you agree.

Comment (roydahan) [reply sdcm/cluster.py]: Yes, 5 seconds makes more sense.

**Pull request merged**